fe1-web/src/features/wallet/objects/transaction/Transaction.ts

```diff
@@ -0,0 +1,387 @@
+    }
+
+    const inputs: Omit<TransactionInputState, 'script'>[] = Transaction.getInputsInToSign(
+      from.publicKey.valueOf(),
+      inputTransactions,
+    );
+
+    // Concatenate the data to sign
+    const dataString = Transaction.concatenateTxData(inputs, outputs);
+
+    // Sign with the popToken
+    const signature = from.privateKey.sign(Base64UrlData.encode(dataString));
+
+    // Reconstruct the txIns with the signature
+    const finalInputs: TransactionInputState[] = inputs.map((input) => {
+      return {
+        ...input,
+        script: {
+          type: STRINGS.script_type,
+          publicKey: from.publicKey.valueOf(),
+          signature: signature.valueOf(),
+        },
+      };
+    });
+
+    return Transaction.fromState({
+      version: 1,
+      inputs: finalInputs,
+      outputs: outputs,
+      lockTime: 0,
+    });
+  }
+
+  /**
+   * Creates a coinbase transaction
+   * @param organizerKP the organizer's key pair
+   * @param to the receiver of the coinbase transaction
+   * @param amount the amount to send
+   */
+  public static createCoinbase(organizerKP: KeyPair, to: PublicKey, amount: number): Transaction {
+    const toPublicKeyHash = Hash.fromPublicKey(to);
+    const outputTo = {
+      value: amount,
+      script: {
+        type: STRINGS.script_type,
+        publicKeyHash: toPublicKeyHash.valueOf(),
+      },
+    };
+
+    const outputs: TransactionOutputState[] = [outputTo];
+
+    const input: Omit<TransactionInputState, 'script'> = {
+      txOutHash: STRINGS.coinbase_hash,
+      txOutIndex: 0,
+    };
+
+    // Concatenate the data to sign
+    const dataString = Transaction.concatenateTxData([input], outputs);
+
+    // Sign with the popToken
+    const signature = organizerKP.privateKey.sign(Base64UrlData.encode(dataString));
+
+    // Reconstruct the inputs with the signature of the organizer
+    const finalInput: TransactionInputState = {
+      ...input,
+      script: {
+        type: STRINGS.script_type,
+        publicKey: organizerKP.publicKey.valueOf(),
+        signature: signature.valueOf(),
+      },
+    };
+
+    return Transaction.fromState({
+      version: 1,
+      inputs: [finalInput],
+      outputs: outputs,
+      lockTime: 0,
+    });
+  }
+
+  /**
+   * Constructs a partial Input object from transaction messages to take as input
+   * @param pk the public key of the sender
+   * @param transactions the transaction messages used as inputs
+   */
+  private static readonly getInputsInToSign = (
+    pk: string,
+    transactions: TransactionState[],
+  ): Omit<TransactionInputState, 'script'>[] => {
+    return transactions.flatMap((tr) =>
+      tr.outputs
+        .filter(
+          (output) => output.script.publicKeyHash.valueOf() === Hash.fromPublicKey(pk).valueOf(),
+        )
+        .map((output, index) => {
+          return {
+            txOutHash: tr.transactionId!.valueOf(),
+            txOutIndex: index,
+          };
+        }),
+    );
+  };
```

Comment:
True, but this does not seem to work here I can't understand why
```suggestion
        .map((output, index) => {
            txOutHash: tr.transactionId!.valueOf(),
            txOutIndex: index,
        }),
```

Reply:
True, but this does not seem to work and I can't understand why
```suggestion
  ): Omit<TransactionInputState, 'script'>[] => {
    return transactions.flatMap((tr) =>
      tr.outputs
        .filter(
          (output) => output.script.publicKeyHash.valueOf() === Hash.fromPublicKey(pk).valueOf(),
        )
        .map((output, index) => {
            txOutHash: tr.transactionId!.valueOf(),
            txOutIndex: index,
        }),
    );
  };
```

Reply:
probably the braces. `() => { x; }` may not be the way.

Reply:
Ahh yes, this is because

`() => {}` is the syntax for a function where a function body is expected in the curly braces. If you want to directly return an object (which also uses the curly brace syntax) you have to add parenthesis:


`() => ({someKey: "someValue"})`

Btw, you can also use this syntax at the top level, so instead of

```
): Omit<TransactionInputState, 'script'>[] => {
return transactions.flatMap((tr) => ...
```

you could write

```
): Omit<TransactionInputState, 'script'>[] => transactions.flatMap((tr) => ...
```

Reply:
That was it thanks !